lib/model/autocomplete.dart

```diff
@@ -461,3 +514,78 @@ class UserMentionAutocompleteResult extends MentionAutocompleteResult {
 // TODO(#233): // class UserGroupMentionAutocompleteResult extends MentionAutocompleteResult {
 
 // TODO(#234): // class WildcardMentionAutocompleteResult extends MentionAutocompleteResult {
+
+class TopicAutocompleteView extends AutocompleteView<TopicAutocompleteQuery, TopicAutocompleteResult, String> {
+  TopicAutocompleteView._({required super.store, required this.streamId});
+
+  factory TopicAutocompleteView.init({required PerAccountStore store, required int streamId}) {
+    final view = TopicAutocompleteView._(store: store, streamId: streamId);
+    store.autocompleteViewManager.registerTopicAutocomplete(view);
+    view._fetch();
+    return view;
+  }
+
+  final int streamId;
+  Iterable<String> _topics = [];
+  bool _isFetching = false;
+
+
+  /// Fetches topics of the current stream narrow, expected to fetch
+  /// only once per lifecycle.
+  ///
+  /// Starts fetching once the stream narrow is active, then when results
+  /// are fetched it restarts search to refresh UI showing the newly
+  /// fetched topics.
+  Future<void> _fetch() async {
+    if (_isFetching) return;
+    _isFetching = true;
+    final result = await getStreamTopics(store.connection, streamId: streamId);
+    _topics = result.topics.map((e) => e.name);
+    _isFetching = false;
+    if (_query != null) _startSearch(_query!);
+  }
+
+  @override
+  Iterable<String> getSortedItemsToTest(TopicAutocompleteQuery query) => _topics;
```

Comment:
Thanks! I've [started a thread](https://chat.zulip.org#narrow/stream/412-api-documentation/topic/Does.20get-stream-topics.20returns.20a.20sorted.20list.20of.20topics.3F/near/1916151)

Reply:
Anyways, I believe neither zulip-web nor zulip-mobile sorts the returned result.

Reply:
(I think you already know this is still open, but bumping just so this remains visible in the new review)

Reply:
I've [tried to verify this](https://chat.zulip.org/#narrow/stream/412-api-documentation/topic/Does.20get-stream-topics.20returns.20a.20sorted.20list.20of.20topics.3F/near/1917883). I think we can now use `_topics` assuming it is already sorted.

Reply:
Cool. Verifying the existing server's behavior isn't enough in itself, because that doesn't guarantee the server won't change behavior later… but that thread now also has Tim's agreement that we intend that as part of the API, and an intention to write it down in the docs. So that means we can rely on it for the future too.